functorch/experimental/_cond.py

```diff
@@ -122,35 +122,8 @@ def cond_dense(pred, true_fn, false_fn, operands):
 @cond.py_impl(DispatchKey.Autograd)
 def cond_autograd(pred, true_fn, false_fn, *operands):
     # TODO: support autograd
-    flat_operands, _ = tree_flatten([true_fn, false_fn] + [operands])
-
-    requires_grad = any(
-        isinstance(arg, torch.Tensor) and arg.requires_grad
-        for arg in flat_operands
-    )
-
-    with _ExcludeDispatchKeyGuard(DispatchKeySet(DispatchKey.AutogradCPU)):
-        result = cond(pred, true_fn, false_fn, *operands)
-
-        # If there is requires_grad, we delay the error until backward pass
-        if requires_grad:
-            # cond can only return one value
-            err_fn = torch._C._functions.DelayedError(
-                b"NYI: torch.cond doesn't support autograd",
-                1,
-            )
-            # Create aliases of the output that has requires_grad=True. We need
-            # at least one of the inputs to err_fn to require grad so that the
-            # output will have a grad_fn.
-
-            def fake_requires_grad(var):
-                if var is not None:
-                    var = var.detach()
-                    var.requires_grad = True
-                return var
-            return err_fn(fake_requires_grad(result))
-
-        return result
+    return autograd_not_implemented(
+        lambda *operands: cond(pred, true_fn, false_fn, *operands), "cond", True, *operands)
```

Comment:
Is it possible to get the UX to look like the following?
```
cond.py_impl(DispatchKey.Autograd)(autograd_not_implemented2(cond, deferred_error=True))
```
The way we'd implement `autograd_not_implemented2` is:
```py
def autograd_not_implemented2(op, deferred_error):
    def inner(*args):
        autograd_not_implemented(op, op.__name__, deferred_error, args)
    return inner
```